grpc.go

```diff
@@ -1320,13 +1321,18 @@ func (a *agentGRPC) ReseedRandomDev(ctx context.Context, req *pb.ReseedRandomDev
 func (a *agentGRPC) GetGuestDetails(ctx context.Context, req *pb.GuestDetailsRequest) (*pb.GuestDetailsResponse, error) {
 	var details pb.GuestDetailsResponse
 	if req.MemBlockSize {
-		data, err := ioutil.ReadFile("/sys/devices/system/memory/block_size_bytes")
-		if err != nil {
-			return nil, err
-		}
-		details.MemBlockSizeBytes, err = strconv.ParseUint(string(data[:len(data)-1]), 16, 64)
-		if err != nil {
-			return nil, err
+		if _, err := os.Stat(sysfsMemoryBlockSizePath); err != nil {
```

Comment:
This code can be simplified - rather than calling `os.Stat()`, just call the `ioutil.ReadFile()`. If an error is returned, check it with [`IsNotExist`](https://golang.org/pkg/os/#IsNotExist).

Also, I think it would be useful if you define a global `var` for `/sys/devices/system/memory/block_size_bytes` (or maybe find a way to pass the file to test as a function arg) as that would allow you to update the test in  `TestGetGuestDetails` to handle both scenarios.

Reply:
> Also, I think it would be useful if you define a global `var` for `/sys/devices/system/memory/block_size_bytes` (or maybe find a way to pass the file to test as a function arg) as that would allow you to update the test in `TestGetGuestDetails` to handle both scenarios.

Yes, I have defined a global variable `sysfsMemoryBlockSizePath` for `/sys/devices/system/memory/block_size_bytes`, and update the test. Is that what you mean?

Reply:
Hi @clarecch - partly. Now that you have that path stored in a variable, the test can change the value so you can test the following scenarios:

- When the file does not exist.
- When the file exists and contains data in the correct format for `ParseUint()`.
- When the file exists and contains data in an invalid format for `ParseUint()`

  You could just write `hello world` to a tempfile and assign `sysfsMemoryBlockSizePath` to that tempfile for example. Yes, this is paranoia ;)

But also note my comment - you do not need a call to `os.Stat()` at all I believe.

Reply:
Thanks for explanation @jodh-intel